arangod/Transaction/Methods.cpp

```diff
@@ -1466,10 +1466,23 @@ OperationResult transaction::Methods::insertLocal(
     ManagedDocumentResult result;
     TRI_voc_tick_t resultMarkerTick = 0;
     TRI_voc_rid_t revisionId = 0;
+    auto needsLock = [this,&collection]() -> bool {return !isLocked(collection, AccessMode::Type::WRITE);};
+
+    Result res = collection->insert( this, value, result, options
+                                   , resultMarkerTick, needsLock(), revisionId
+                                   );
 
-    Result res =
-        collection->insert(this, value, result, options, resultMarkerTick,
-                           !isLocked(collection, AccessMode::Type::WRITE), revisionId);
+    if(options.overwrite && res.is(TRI_ERROR_ARANGO_UNIQUE_CONSTRAINT_VIOLATED)){
+      // RepSert Case - unique_constraint violated -> maxTick has not changed -> try replace
```

Comment:
could be a unique constraint violation on a secondary index. does this need to be differentiated?